src/world.rs

```diff
@@ -295,6 +297,10 @@ impl<'a> Join for &'a Entities {
     }
 }
 
+// TODO: Is this correct?
+#[cfg(feature="parallel")]
+impl<'a> ParJoin for &'a Entities {}
```

Comment:
I didn't look at the rest of the code yet, but sharing an immutable struct always is safe.